### PR TITLE
Upgrade to scriptworker 41.0.0 and latest iscript version

### PR DESCRIPTION
**Pull request by bhearsum**

### CODE REVIEW

**Review by escapewindow — CHANGES_REQUESTED** [review]
idna has a major version bump, but the major blocker is tc should be >=40.

**Comment by bhearsum** [comment]
> idna has a major version bump, but the major blocker is tc should be >=40.

Fixed the taskcluster issue. The idna major version bump is just for dropping python 2 support, no API changes AFAICT

File: modules/signing_worker/files/requirements.dep.txt [hunk 1]
```diff
@@ -41,129 +41,178 @@ aiohttp==3.7.4.post0 \
...
-    --hash=sha256:c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a \
-    # via -r https://raw.githubusercontent.com/mozilla-releng/scriptworker-scripts/0ac27f13cc443cdadbaa715442ee720aec1da7a7/scriptworker_client/requirements/base.in, -r https://raw.githubusercontent.com/mozilla-releng/scriptworker/a76a16da95e0d2483907b8a327b7a3f69743fbff/requirements.txt
-jwcrypto==0.8 \
-    --hash=sha256:16e17faa4dce36551ade3a3ccb06236a61e5924ea1db163c9be9827acf935a82 \
-    --hash=sha256:b7fee2635bbefdf145399392f5be26ad54161c8271c66b5fe107b4b452f06c24 \
+    --hash=sha256:c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a
+    # via
+    #   -r https://raw.githubusercontent.com/mozilla-releng/scriptworker-scripts/4626000ccc5e264c69b6f5f06f67a7ed6362be14/scriptworker_client/requirements/base.in
+    #   -r https://raw.githubusercontent.com/mozilla-releng/scriptworker/cd731fa4b02b10b430f9414df678f47640491931/requirements.txt
+jwcrypto==1.0 \
```
Comment (escapewindow) [inline modules/signing_worker/files/requirements.dep.txt:198]: Nothing's jumping out at me: https://github.com/latchset/jwcrypto/releases/tag/v1.0.0

**Review by escapewindow — DISMISSED** [review]
Thanks!
Given yesterday, we may want to a) quarantine a prod mac, b) ssh in, c) disable the periodic puppet launch daemon, d) manually puppetize against master (I tend to edit /usr/local/bin/run-puppet.sh), e) possibly manually run script with `mac_notarize` behavior to get one round trip sign+notarize+poll+staple run, clean up work/ and artifacts/ and restart the periodic puppet launch daemon, f) unquarantine, and then g) push master to production-mac-signing to roll out everywhere.

(Yolo is push to production-mac-signing with some non-zero chance of needing Relops to help us rescue the pool.)

**Comment by bhearsum** [comment]
In testing this I discovered that `notarization_poller` calls `claimWork` itself and needed a patch: https://github.com/mozilla-releng/scriptworker-scripts/pull/395

This PR will need an update after that one is merged.

**Comment by bhearsum** [comment]
Planning to roll this out tomorrow.

**Review by escapewindow — APPROVED** [review]

File: modules/signing_worker/files/requirements.dep.txt (hunk 1 above)
Comment (escapewindow) [reply modules/signing_worker/files/requirements.dep.txt:198]: Oops, I thought I stamped it at this point. Sorry!

**Pull request merged**